### PR TITLE
Add C++ Frontend : triplet-margin-loss

### PR DESCRIPTION
**Pull request by PyExtreme**
Hi @yf225 , Here is the **C++ frontend API TripletMarginLoss** implementation. I have got some branching issues and therefore am opening a pull request.

The link of our previous conversation is here and I have made all the changes as mentioned.
https://github.com/pytorch/pytorch/pull/27525

### CODE REVIEW

File: test/cpp/api/modules.cpp [hunk 1]
```diff
@@ -1294,6 +1308,12 @@ TEST_F(ModulesTest, PrettyPrintCosineSimilarity) {
       "torch::nn::CosineSimilarity(dim=0, eps=0.5)");
 }
 
+TEST_F(ModulesTest, PrettyPrintTripletMarginLoss) {
+  ASSERT_EQ(
+      c10::str(TripletMarginLoss(TripletMarginLossOptions().margin(3).p(2).eps(1e-06).swap(false))),
+      "torch::nn::TripletMarginLoss(margin=3, p=2, eps=1e-06, swap=false)");
```
Comment (yf225) [inline test/cpp/api/modules.cpp]: Is it supposed to also print `reduction`? I guess we don't have a nice string representation of the reduction enum at the moment, and please feel free to not print it for now. :D

**Review by yf225 — COMMENTED** [review]
Thanks a lot for the great work @PyExtreme! I left some comments. I believe we should also rebase this PR on top of current master and fix the code conflicts.

File: test/cpp/api/modules.cpp (hunk 1 above)
Comment (PyExtreme) [reply test/cpp/api/modules.cpp]: @yf225 , no this won't be printing `reduction` here.

Comment (PyExtreme) [reply test/cpp/api/modules.cpp]: @yf225 , Thanks a lot for reviewing it. I have learnt a lot from it, gained confidence and got familiar with the pytorch codebase.
Will surely make the mentioned changes and fix the conflicts.

Thank you

File: torch/csrc/api/src/nn/modules/loss.cpp [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #include <torch/nn/modules/loss.h>
-
+<<<<<<< HEAD
+=======
+using namespace std
+>>>>>>> Add C++ Frontend - triplet-margin-loss
```
Comment (yf225) [inline torch/csrc/api/src/nn/modules/loss.cpp]: We need to fix the code conflict here, and we also shouldn't do `using namespace std`.

File: torch/csrc/api/src/nn/modules/loss.cpp [hunk 3]
```diff
@@ -32,5 +35,30 @@ Tensor HingeEmbeddingLossImpl::forward(
...
+
+void TripletMarginLossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::TripletMarginLoss(margin=" << options.margin() << 
+            ", p=" << options.p() <<
+            ", eps=" << options.eps() << std::boolalpha <<
+<<<<<<< HEAD
+            ", swap=" << options.swap() <<
+=======
+            ", swap=" <<options.swap() <<
+>>>>>>> Add C++ Frontend - triplet-margin-loss
```
Comment (yf225) [inline torch/csrc/api/src/nn/modules/loss.cpp]: We need to fix the code conflict here.

File: torch/csrc/api/src/nn/modules/loss.cpp [hunk 4]
```diff
@@ -32,5 +35,30 @@ Tensor HingeEmbeddingLossImpl::forward(
...
+            ", reduction=" << options.reduction() << ")";
```
Comment (yf225) [inline torch/csrc/api/src/nn/modules/loss.cpp]: It seems confusing that we have logic for printing `reduction` here, but the actual `pretty_print` string doesn't have `reduction`. I suggest removing the `reduction` printing logic here for now, until we have better support for printing named enums :)

**Review by yf225 — COMMENTED** [review]
Thanks @PyExtreme! I left some comments.

**Comment by PyExtreme** [comment]
Hi @yf225 , I had to create a new branch to tackle merge conflict since I am using cloud due to some limitations on my PC. Therefore, I don't have enough command there.

Also, I have incorporated the changes you have put before here

Here is the link to my new PR
https://github.com/pytorch/pytorch/pull/27713

From next time, I will find a smoother method for fixing conflicts.

Thank you

File: torch/csrc/api/src/nn/modules/loss.cpp (hunk 4 above)
Comment (PyExtreme) [reply torch/csrc/api/src/nn/modules/loss.cpp]: Hi @yf225 , I mistakenly pushed these changes. Sorry for wasting your time, I was still working on the PR.

Comment (PyExtreme) [reply torch/csrc/api/src/nn/modules/loss.cpp]: Here is my updated PR #27713 

Thank you

**Pull request closed without merging**